src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -492,6 +506,20 @@ private void ProcessEvents()
                                 }
                             }
                             expandedName = associatedDirectoryEntry.GetPath(true, nextEvent.name);
+
+                            // inotify picks up changes to symlinks and reports them as file changes, which breaks the contract. We need
```

Comment:
How does that break the contract?  Aren't we excluding symlink events with the IN_DONT_FOLLOW flag, such that if an event does occur on a symlink, it's for the link file itself rather than for the target, and in that case, isn't it a valid event?